### PR TITLE
icon-selector with material-ui and redux

### PR DESCRIPTION
**Pull request by PiotrKukuc12**
I had to export icons object from FlowNodeIcons, I belive there is better way to make this work without duplicating

### CODE REVIEW

**Comment by atompie** [comment]
@PiotrKukuc12 I was going to ask you about it. I will take a look at this and will get back to you￼

File: src/components/elements/IconSelector.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const open = Boolean(anchorEl);
+  const id = open ? 'simple-popover' : undefined;
+
+  return (
```
Comment (atompie) [inline src/components/elements/IconSelector.js:22]: The component has not all the functionality described in the task. 

* There is no input field that the icon name should be copied. 
* Positioning of the popover is not the same as on the wire-frame

File: src/components/flow/FlowNodeIcons.js [hunk 2]
```diff
@@ -153,4 +153,78 @@ export default function FlowNodeIcons({icon, size=20}) {
     }
 
     return renderIcon()
-}
\ No newline at end of file
+}
+
+let size = 20
+export const icons = {
```
Comment (atompie) [inline src/components/flow/FlowNodeIcons.js:159]: Icons that exist in the FlowNodeIcons function  can be moved outside the function:

```
export const icons = {
        "store": <FiDatabase size={size}/>,
        "json": <VscJson size={size}/>,
        "split": <BiGitRepoForked size={size} style={{ transform: "rotateX(180deg)"}}/>,
        "join": <BiGitRepoForked size={size} />,
...
}

export default function FlowNodeIcons({icon, size=20}) {
....
```
This was the list can be exported and also used in the FlowNodeIcons. This solves the problem of duplicated icon list.

File: src/redux/reducers/iconSlice.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { createSlice } from '@reduxjs/toolkit';
```
Comment (atompie) [inline src/redux/reducers/iconSlice.js:1]: Why did you use redux for such a trival operation as adding icon. Icon as described in task should be copied to input field that is a part of this component. Then onChange property could be set to notify any outer component about the change. 

Using redux has its drawbacks. It always re-renders the whole application. Therefore it should be used with caution only for cross component actions.

File: src/redux/store.js [hunk 4]
```diff
@@ -19,7 +20,8 @@ export default configureStore({
             alertReducer,
             eventDataGridReducer,
             uqlReducer,
-            pagingReducer
+            pagingReducer,
```
Comment (atompie) [inline src/redux/store.js:23]: See comment above.

**Review by atompie — CHANGES_REQUESTED** [review]
Please add input field and make it act as described in the task. Thanks.

**Comment by PiotrKukuc12** [comment]
I have clearly misunderstood task. I thought it might be good idea to add an icon that saves the name to clipboard, let me know if it is a bad idea, then i'll remove it. Hope it now works as it should.

**Pull request merged**